### PR TITLE
Build restic at container build time

### PR DESCRIPTION
**Pull request by jinnko**
What is the purpose of this change? What does it change?
--------------------------------------------------------

Using docker's multi-stage builds we can build the restic/rest-server
within a golang build environment then create a container for use
(without the build environment) in a second build stage.

The advantages are:

1. Building the rest-server is predictable in a pristine environment
   each time.
2. Container builds ensure we get the latest rest-server every time.

Was the change discussed in an issue or in the forum before?
------------------------------------------------------------

~~Not yet - will create an issue now.~~

https://github.com/restic/rest-server/issues/146

Checklist
---------

NOTE: I will go through this checklist as soon as possible to satisfy all the requirements.

- [x] I have enabled [maintainer edits for this PR](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
- [ ] I have added tests for all changes in this PR
- [x] I have added documentation for the changes (in the manual)
- [x] There's a new file in `changelog/unreleased/` that describes the changes for our users (template [here](https://github.com/restic/rest-server/blob/master/changelog/TEMPLATE))
- [ ] I have run `gofmt` on the code in all commits
- [x] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/rest-server/commits/master)
- [x] I'm done, this Pull Request is ready for review

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,11 +19,15 @@ The required version of restic backup client to use with `rest-server` is [v0.7.
...
+
 ## Docker
 
 ### Build image
 
-Put the `rest-server` binary in the current directory, then run:
+The docker image build process will build a fresh version of the rest-server and package that into
+a usable container.
+
+TO build the binary along with the container run:
```
Comment (MichaelEischer) [inline README.md:30]: `s/TO/To`

File: Dockerfile [hunk 2]
```diff
@@ -1,3 +1,14 @@
+FROM golang:alpine
```
Comment (MichaelEischer) [inline Dockerfile:1]: Please name the build container using `AS builder`.

File: README.md (hunk 1 above)
Comment (jinnko) [reply README.md]: Fixed.

File: Dockerfile (hunk 2 above)
Comment (jinnko) [reply Dockerfile]: Done

**Comment by jinnko** [comment]
Is there anything further you'd like to see in this PR before it's good to go?

**Review by MichaelEischer — APPROVED** [review]
LGTM. I've slightly tweaked the changelog entry.

**Pull request merged**